packages/live-share-media/src/VolumeLimiter.ts

```diff
@@ -3,59 +3,64 @@
  * Licensed under the Microsoft Live Share SDK License.
  */
 
- import { TimeInterval } from '@microsoft/live-share';
- import { IMediaPlayer } from './IMediaPlayer';
+import { TimeInterval } from '@microsoft/live-share';
+import { IMediaPlayer } from './IMediaPlayer';
 
 export enum LevelType { fixed, percentage }
 
+/**
+ * Smooth audio level changes when selectedVolume is modified, or if volume limiting has started/ended.
+ */
 export class VolumeLimiter {
     private readonly _player: IMediaPlayer;
+    private readonly _volumeChangeDuration = new TimeInterval(500);
 
-    // Settings
-    private readonly _duration = new TimeInterval(2000);
-    private readonly _rampDownDuration = new TimeInterval(500);
-    private readonly _rampUpDuration = new TimeInterval(500);
-    private _level: number = 0.1
+    private _selectedVolume = 1.0;
+    private _limited = false;
+    private _level = 0.1
     private _levelType: LevelType = LevelType.fixed;
-
-    // Running state
-    private _running = false;
-    private _endTime = 0;
     private _startTime = 0;
     private _startVolume = 0;
-    private _inRamp = false;
+    private _running = false;
 
     constructor(player: IMediaPlayer) {
         this._player = player;
+        this._player.volume = this._selectedVolume;
```

Comment:
>this._player.volume = this._selectedVolume; [](http://example.com/codeflow?start=8&length=43)

Do we really want to always set the players volume to 1.0? I don't think we should tamper with the players volume at all until activated.

Reply:
changed to set selected volume to whatever the current player volume is